src/contract.rs

```diff
@@ -73,9 +76,114 @@ pub fn execute(
             validator_oper_addr,
         } => remove_validator(deps, info, validator_oper_addr),
         ExecuteMsg::WithdrawFunds { amount } => withdraw_funds(deps, info, amount),
+
+        ExecuteMsg::DeleteMetricsForTimestamp {
+            timestamp,
+            validator_idx,
+            validator_ct,
+        } => delete_metrics_for_timestamp(
+            deps,
+            info,
+            timestamp,
+            validator_idx as usize,
+            validator_ct as usize,
+        ),
+
+        ExecuteMsg::DeleteMetricsForValidator {
+            validator_opr_addr,
+            timestamp_idx,
+            timestamp_ct,
+        } => delete_metrics_for_validator(
+            deps,
+            info,
+            validator_opr_addr,
+            timestamp_idx as usize,
+            timestamp_ct as usize,
+        ),
+
+        ExecuteMsg::RemoveTimestamp { timestamp_idx } => remove_timestamp(deps, info, timestamp_idx)
     }
 }
 
+//todo; write test for this
+//note: this is also inefficient, potentially O(T), due to array deletion. not sure if
+// cosmwasm library optimizes this.
+fn remove_timestamp(
+    deps: DepsMut,
+    info: MessageInfo,
+    timestamp: u64) -> Result<Response, ContractError> {
+
+    // can only be called by manager
+    if !sender_is_manager(&deps, &info) {
+        return Err(ContractError::Unauthorized {});
+    }
+
+    let mut state = STATE.load(deps.storage)?;
+    let existing_timestamps_length = state.cron_timestamps.len();
+
+    let new_timestamps = state.cron_timestamps.into_iter()
+        .filter(|state_tstamp| {state_tstamp.ne(&timestamp)})
+        .collect::<Vec<u64>>();
+
+    if new_timestamps.len() == existing_timestamps_length {
```

Comment:
As a general code paradigm, keep in mind to not "early return". 

I have noticed this tends to create bugs especially when some part of the state (not state struct) needs to be updated.  Until we find out a better pattern, I would advice not to early return. 

It's fine here because STATE is the only update.